### PR TITLE
ci(*): automate the version bump process

### PR DESCRIPTION
**Pull request by Mossaka**
this PR automates the version bump process to
ask the bot to create a PR to bump the respective
crate's version to the given one on behalf of humans.

the PR created by the bot contains magic message that
will be parsed by another workflow `publish` which triggers
on merge to the main branch.

The `publish` workflow parses the commit message to get
crate, version and dry-run information to be able to
publish the crate, release binary and push tags.

the motivation for this commit is to further simplify the
release process

builds on top of #548

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -73,152 +58,48 @@ jobs:
...
+          ${{ inputs.crate }} v${{ inputs.version }} when merged. The commit
+          message for this PR has a marker that is detected by CI to create
+          tags and publish crate artifacts.
+
+          When first opened this PR will not have CI run because it is generated
+          by a bot. A maintainer should close this PR and then reopen it to
+          trigger CI to execute which will then enable merging this PR.
+          EOF
+      - name: Make a PR
+        run: gh pr create -B "$PR_BASE" -H "$PR_HEAD" --title "$PR_TITLE" --body "$(cat ./pr-body)"
```
Comment (cpuguy83) [inline .github/workflows/release.yml:103]: So I recalled the problem with using the github actions token for opening pull requests: Jobs will not trigger on that pull request.
So you can open the PR but CI will not run.

Comment (Mossaka) [reply .github/workflows/release.yml:103]: Ahh yes, but I think that's fine as long as we document it in the release.md something like "you need to close the PR and re-open the PR to trigger the CIs". How does this sound?

Comment (Mossaka) [reply .github/workflows/release.yml:103]: "When the PR opens, close it then reopen it. Don't ask questions."

**Review by devigned — DISMISSED** [review]
lgtm

File: .github/workflows/release.yml (hunk 1 above)
Comment (cpuguy83) [reply .github/workflows/release.yml:103]: If that works that's fine.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
+
+name: Publish Artifacts
+
+on:
+  push:
+    branches: [main]
```
Comment (cpuguy83) [inline .github/workflows/publish.yml:6]: This doesn't seem right.
Looks like its going to run this on every merge to main?

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 2 above)
Comment (Mossaka) [reply .github/workflows/publish.yml:6]: Perhaps, a release branch would make more sense?

Comment (cpuguy83) [reply .github/workflows/publish.yml:6]: It seems like the intent would be to run this on tag rather than on a branch.

Comment (Mossaka) [reply .github/workflows/publish.yml:6]: The intention is to run the `publish` action after the version bump PR merged to `main` branch

**Comment by Mossaka** [comment]
This PR may need some love, but let me first release a new version of runwasi

**Comment by jsturtevant** [comment]
Going to close this up, but feel free to refresh at some point

**Pull request closed without merging**